### PR TITLE
added: support for the dörfel marking strategy

### PR DESCRIPTION
**Pull request by akva2**
mark elements/functions which contributes beta% to the total error

### CODE REVIEW

File: src/SIM/AdaptiveSIM.C [hunk 1]
```diff
@@ -418,6 +420,17 @@ bool AdaptiveSIM::adaptMesh (int iStep, std::streamsize outPrec)
...
     break;
+  case DORFEL:
+  {
+    for (const DblIdx& error : errors)
+      sumErr += error.first;
+    double total = 0.0;
+    size_t i = 0;
+    while (total < sumErr*beta/100.0)
+      total += errors[i++].first;
+    limit = errors[i].first;
```
Comment (kmokstad) [inline src/SIM/AdaptiveSIM.C]: Cosmetic alternative to lines 428-431:
```
sumErr *= beta/100.0;
for (const DblIdx& error : errors)
  if (total < sumErr)
    total += error.first;
  else
  {
    limit = error.first;
    break;
  }
```
and we don't need the brace-pair after the case-statement, if the `total`variable is declared outside together with `limit`and `sumErr`.

Comment (akva2) [reply src/SIM/AdaptiveSIM.C]: i'll do the rewrite, but if i make total non-local i will change name. it's too easily confused with all sorts of things outside the local context.

Comment (kmokstad) [reply src/SIM/AdaptiveSIM.C]: sure, looks good :-) But maybe also initialize limit to zero (or maybe `errors.back().first`then is the correct value) before the loop, just to cover the unlikely event of a very big beta value such that curErr never gets above sumErr

Comment (akva2) [reply src/SIM/AdaptiveSIM.C]: done.

**Comment by akva2** [comment]
test added in https://github.com/OPM/IFEM-Poisson/pull/28

merging.

**Pull request merged**